decaylanguage/dec/dec.py

```diff
@@ -87,33 +90,78 @@ def __init__(self, filename):
         self._grammar = None       # Loaded Lark grammar definition file
         self._grammar_info = None  # Name of Lark grammar definition file
 
-        # Conversion to handle pathlib on Python < 3.6:
-        self._dec_file_name = str(filename)  # Name of the input decay file
-        self._parsed_dec_file = None      # Parsed decay file
+        # Name(s) of the input decay file(s)
+        self._dec_file_names = None
+        self._dec_file = None
+        if filename:
+            filename = str(filename)  # Conversion to handle pathlib on Python < 3.6
+
+            # Check input file
+            if not os.path.exists(filename):
+                raise FileNotFoundError("'{0}'!".format(filename))
+
+            self._dec_file_names = filename
+            self._dec_file = open(self._dec_file_names).read()
 
-        self._parsed_decays = None  # Particle decays found in the decay file
+        self._parsed_dec_file = None  # Parsed decay file
+        self._parsed_decays = None    # Particle decays found in the decay file
 
         # By default, consider charge-conjugate decays when parsing
         self._include_ccdecays = True
 
     @classmethod
-    def from_file(cls, filename):
+    def from_string(cls, filecontent):
         """
-        Parse a .dec decay file.
+        Parse a .dec decay file provided as a multi-line string.
 
         Parameters
         ----------
-        filename: str
-            Input .dec decay file name.
+        filecontent: str
+            Input .dec decay file content.
         """
-        # Conversion to handle pathlib on Python < 3.6:
-        filename = str(filename)
+        stream = StringIO(filecontent)
+        stream.seek(0)
+
+        _cls = cls()
+        _cls._dec_file_names = '<dec file input as a string>'
+        _cls._dec_file = stream.read()
 
-        # Check input file
-        if not os.path.exists(filename):
-            raise FileNotFoundError("'{0}'!".format(filename))
+        return _cls
 
-        return cls(filename)
+    @classmethod
+    def from_files(cls, filenames):
```

Comment:
Couldn't either a) call the base constructor here with each file and merge, or b) allow a list of decay files in the base constructor?

Reply:
I've removed this class constructor making the default one versatile.